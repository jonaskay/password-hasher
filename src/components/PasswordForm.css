.PasswordForm {
  margin-bottom: 2rem;
}

.PasswordForm-field {
  margin-bottom: 1rem;
}

.PasswordForm-field > label, .PasswordForm-field > input {
  display: inline-block;
}

.PasswordForm-field > label {
  width: 27%;
}

.PasswordForm-field > input {
  border: none;
  border-radius: .25rem;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.10);
  box-sizing: border-box;
  color: rgba(0,0,0,.87);
  padding: .5rem;
  width: 73%;
}

.PasswordForm-button {
  background: #ffd43b;
  border: none;
  border-radius: .25rem;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.12), 0 2px 4px 0 rgba(0,0,0,0.08);
  color: rgba(0,0,0,.87);
  cursor: pointer;
  font-weight: 500;
  padding: .5rem;
  width: 73%;
}
