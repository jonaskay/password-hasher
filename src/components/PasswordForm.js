import React, { Component } from 'react';
import './PasswordForm.css';

class PasswordForm extends Component {
  render() {
    return (
      <form className="PasswordForm" onSubmit={this.props.onSubmit}>
        <div className="PasswordForm-field">
          <label>Domain</label>
          <input
            type="text"
            name="domain"
            placeholder="www.example.org"
            value={this.props.domain}
            onChange={this.props.onChange}
          />
        </div>
        <div className="PasswordForm-field">
          <label>Password</label>
          <input
            type="password"
            name="password"
            value={this.props.password}
            onChange={this.props.onChange}
          />
        </div>
        <div style={{ marginTop: '1.25rem', textAlign: 'right' }}>
          <input className="PasswordForm-button" type="submit" value="Generate" />
        </div>
      </form>
    );
  }
}

export default PasswordForm;
