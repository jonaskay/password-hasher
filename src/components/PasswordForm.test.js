import React from 'react';
import { shallow } from 'enzyme';
import PasswordForm from './PasswordForm';

describe('PasswordForm', () => {
  it('renders correctly', () => {
    const shallowed = shallow(<PasswordForm />);
    expect(shallowed).toMatchSnapshot();
  });
});
