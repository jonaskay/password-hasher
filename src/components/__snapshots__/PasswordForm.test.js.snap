// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForm renders correctly 1`] = `
<form
  className="PasswordForm"
>
  <div
    className="PasswordForm-field"
  >
    <label>
      Domain
    </label>
    <input
      name="domain"
      type="text"
    />
  </div>
  <div
    className="PasswordForm-field"
  >
    <label>
      Password
    </label>
    <input
      name="password"
      type="password"
    />
  </div>
  <div
    style={
      Object {
        "marginTop": "1.25rem",
        "textAlign": "right",
      }
    }
  >
    <input
      className="PasswordForm-button"
      type="submit"
      value="Generate"
    />
  </div>
</form>
`;
